dialoptions.go

```diff
@@ -374,8 +374,8 @@ func init() {
 // is returned by f, gRPC checks the error's Temporary() method to decide if it
 // should try to reconnect to the network address.
 //
-// Deprecated: use WithContextDialer instead.  Will be supported throughout
-// 1.x.
+// Deprecated: use WithContextDialer or DialContext instead of Dial. Will be
```

Comment:
This is incorrect.  `WithContextDialer` is to be used instead of `WithDialer` itself.  I don't think any change here is necessary.

Reply:
You are right. I finally found a working way to port from the deprecated api.

Reply:
Actually this is still incorrect. If you port from grpc.Dial() to grpc.DialContext you don't need WithContextDialer() at all.

Reply:
`WithContextDialer` is the new version of `WithDialer`.  `DialContext` is the new version of `Dial` (though `Dial` itself is not deprecated since a context isn't useful without `WithBlock`).  These are entirely orthogonal.